### PR TITLE
Add build step to generate and compile all templates

### LINKED ISSUE #95: Add build step to compile all templates
We should create and build a templates in the regular build to test, whether they at least compile.
We could add a shell script that creates all extensions (they are then also automatically built)
@vince-fugnitto WDYT?

### PR DESCRIPTION
**Pull request by JonasHelming**
Added a build step that generated and compiles a templates and fails the build if any template fails to build.
How to test: Introduce a change in a template that fails its build.
fixed #95 
Signed-off-by: Jonas Helming <jhelming@eclipsesource.com>

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
I created a test pull-request to verify if the workflow correctly fails https://github.com/eclipse-theia/generator-theia-extension/pull/101.
I'll check if it fails the build, and if multiple templates fail it will display their results (and not fail fast).

File: .github/workflows/generateAndBuildAllTemplates.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+npm install -g yo
```
Comment (vince-fugnitto) [inline .github/workflows/generateAndBuildAllTemplates.sh:1]: The name of this file should be improved for consistency (we usually use `-`).
Example: `frontend-application.ts`, `ci-cd.yaml`.

**Comment by JonasHelming** [comment]
> I created a test pull-request to verify if the workflow correctly fails #101.
> I'll check if it fails the build, and if multiple templates fail it will display their results (and not fail fast).

done!

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -35,3 +35,7 @@ jobs:
       - name: Test
         shell: bash
         run: yarn test
+
+      - name: Generate templates
```
Comment (vince-fugnitto) [inline .github/workflows/ci-cd.yml:39]: minor: should we write:

```suggestion
      - name: Generate and Build Templates
```

**Review by vince-fugnitto — APPROVED** [review]
The changes look good to me, I verified it with #101 :+1: 

I confirmed that the new workflow step does not fail fast, and displays all relevant information.
I verified by producing errors in the `hello-world` and `widget` templates:

[Same Workflow](https://github.com/eclipse-theia/generator-theia-extension/runs/2575439766?check_suite_focus=true):

image

image

**Comment by vince-fugnitto** [comment]
@JonasHelming do you think we would benefit from running the CI (including generation and building) as part of a cron job?
We can potentially identify breaking issues earlier.

**Comment by JonasHelming** [comment]
Good idea, will do: https://github.com/eclipse-theia/generator-theia-extension/issues/102

**Pull request merged**